User edit

Egoes98 opened the pull request:
In this branch are the methods that allow a Guest to change the user data. In the panel for searching hotels for the Guest it is located a button that allows Guest to update the user data they have. It shows a window like the one for registering and the guest writtes down the new data in the fields he wants to change.

Once you summit the changes it calles the server where in the database makes the changes you requested and return a UserDTO updated. In the LoggedUsers the UserDTO is replaceed with the updated one in server side and in client side.

Egoes98 committed:
DAO,VIEW,CONTROLLER,FACADE changes

I have added the view of eddit that is a copy of the one iñigo did , is the register but i will make some changes to make it an user eddit window. i have added the edit method in the controller and remote facade and also the DAO method.

Egoes98 committed:
Localization and DTO

I have added the localization for the messages on the eddit user. Also i have added all the data from user to the DTO, this is what i thought to be able to see all the data on the GUI but we clould change it if you think is not the best way of handlelinh this. This could be a problem because Guest and Admin has different data.

Egoes98 committed:
Test

Added a test method in the UserDAOTest that test the editUserMethod. Is not completly working becaus esome changes has to be made on the method in the DAO.

Egoes98 committed:
Clean code

I cleaned the extra code i did that wasnt good at all. I should she how to do to show the Users his actual data because you cant get that data from the DTO. So maybe server sending the data without DTO?.

Egoes98 committed:
Validation

Change some validation onptions and clean up the code.

Egoes98 committed:
Merge branch 'master' into User-Edit

Egoes98 committed:
Added Refresh

As Iñigo said I added the refresh method  . Also added some documentation. Once GuestWindow is on i will added to it.

iker-barriocanal committed:
user dao to guests edit account data

added a method in IUserDAO and UserDAO to update the data of guests

iker-barriocanal committed:
Added guest profile edit information to the façade and updated userdao

iker-barriocanal committed:
UserDAOTest moved to its directory

iker-barriocanal committed:
FIX small bug, committing transaction twice

Egoes98 committed:
Merge branch 'master' into User-Edit

Egoes98 committed:
Make the methods that handles user updates

The methods for update user finished. Now must see if it works.

Egoes98 committed:
View

Chnaged a line that was for testing.

Egoes98 committed:
Merged Master

Egoes98 committed:
View Added

Added the view to the panels. Edition works. Test must be done.

Egoes98 committed:
Test

Test methods for updateGuest.

iapellaniz commented on src/main/resources/localization_en_US.properties:
```diff
@@ -62,10 +62,30 @@ login.label.password = Password:
 locale.heading = <html><b>Locale settings</b><br>Please select your desired language:</html>
 locale.save = Save changes
 
+<<<<<<< HEAD
```
Please remove these conflict marks

iapellaniz commented on src/main/java/es/deusto/spq/client/gui/views/auth/EditUserView.java:
```diff
@@ -0,0 +1,332 @@
+
+	@Override
+	public ViewPermission getViewPermission() {
+		return ViewPermission.NOT_LOGGED_IN;
```
Should use `LOGGED_IN`, since we're editing our own user

iapellaniz replied:
After testing, it seems that by changing this the window doesn't show.

Please set it to `NONE` and add a TODO comment so that when I merge #28 I can implement the proper permission level

iapellaniz commented on src/main/java/es/deusto/spq/client/gui/views/auth/LoginView.java:
```diff
@@ -27,8 +41,8 @@
     private final JPanel panel_3 = new JPanel();
-    private Logger log;
-    private HotelManagementController controller;
+    private final Logger log;
```
Why the sudden use of finals?

Egoes98 replied:
Yeah sorry about that it was because i had the eclipse configured to automaticly use final when posible. i used it for doing other project and i didnt notice to change that.

iapellaniz commented on src/main/resources/localization_en_US.properties:
```diff
@@ -21,7 +21,7 @@ menu.about.body = <html>\
 
 # Register window
-register.title = Register:
+register.title = Change User info:
```
Missed this one! Please set it back to "Register"

iapellaniz reviewed (changes requested):
- EditUserView.java
    - Please implement the `dispose` method

Egoes98 committed:
RequestedChanges(Read description Please)

I ahve changes what was requested. I also added a localization that was not added to the file. The thing about the dispose method, i have added to the class but how should i use it apart from adding the methods to the view. Every time i want to dispose a frame should i call this method instead of doing frame.dispose ???

Egoes98 commented:
I have put it on the description of the commit but I will also leave a comment. I have added de dispose method but I have only put it on the EditUserView class. I don't know when should I use it, should I change the frame.dispose and put the dispose method call instead?

iapellaniz commented:
> I have put it on the description of the commit but I will also leave a comment. I have added de dispose method but I have only put it on the EditUserView class. I don't know when should I use it, should I change the frame.dispose and put the dispose method call instead?

Don't worry too much about it. You've implemented it well in 2423e2baac4e011766ad5681bba338136cece0f8

iapellaniz commented on src/main/resources/localization_en_US.properties:
```diff
@@ -62,6 +62,24 @@ login.label.password = Password:
 login.register = Register
 
+# EditUser window
+editUser.title = Update User Data:
```
This key is not being used. Please remove it

Egoes98 committed:
Removed

Egoes98 commented:
I have noticed that the view only shows up with the permision NOT_LOGGED_IN, are the others implemented in this branch or you have done them on the session sistem one??. Is to know if its because i have done something wrong. Thanks in advance. If its okay it should be fine.

iapellaniz commented:
I think with NONE should work, but if you cant get it to show just keep it
as it was originally. Sorry for the confusion

On Thu, 16 May 2019 at 20:42, Egoitz <notifications@github.com> wrote:

> I have noticed that the view only shows up with the permision
> NOT_LOGGED_IN, are the others implemented in this branch or you have done
> them on the session sistem one??. Is to know if its because i have done
> something wrong. Thanks in advance. If its okay it should be fine.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/BSPQ18-19/BSPQ19-E2/pull/45?email_source=notifications&email_token=AD4JJRGRA2NENSEVN3PLK2LPVWTJ5A5CNFSM4HMKSI7KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODVSWW7Q#issuecomment-493185918>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD4JJRCOUNUSGNWAZ4VFBALPVWTJ5ANCNFSM4HMKSI7A>
> .
>
-- 
- Iñigo Apellániz Cengotitabengoa

Egoes98 committed:
Permision NONE

I have chnaged the permision to none. I have put a TODO comento to change the permision when LOGGED_IN_GUEST works.

Egoes98 commented:
> I think with NONE should work, but if you cant get it to show just keep it as it was originally. Sorry for the confusion
> On Thu, 16 May 2019 at 20:42, Egoitz ***@***.***> wrote: I have noticed that the view only shows up with the permision NOT_LOGGED_IN, are the others implemented in this branch or you have done them on the session sistem one??. Is to know if its because i have done something wrong. Thanks in advance. If its okay it should be fine. — You are receiving this because you commented. Reply to this email directly, view it on GitHub <#45?email_source=notifications&email_token=AD4JJRGRA2NENSEVN3PLK2LPVWTJ5A5CNFSM4HMKSI7KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODVSWW7Q#issuecomment-493185918>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AD4JJRCOUNUSGNWAZ4VFBALPVWTJ5ANCNFSM4HMKSI7A> .
> -- - Iñigo Apellániz Cengotitabengoa

Yes with NONE works fine also, i have set it to NONE.

iapellaniz reviewed (changes requested):
Just as finishing touches:

- It would be nice if the UI was pre-filled with the current user's data
- Maybe you could allow users to change their data but if they don't enter their password we suppose they don't want to change it

iapellaniz commented:
Also, another thing I've noticed: we should remove the "Don't Save Changes" button — it's redundant (that's what the close button is for anyways)

Egoes98 committed:
Button removed

Egoes98 commented:
> Just as finishing touches:
> 
> * It would be nice if the UI was pre-filled with the current user's data
> * Maybe you could allow users to change their data but if they don't enter their password we suppose they don't want to change it

Ok, button removed.
About displaying the current user's data i also think it would be relly nice, in fact it was thought to be that way. The thing is that the UserDTO doesnt have all the user data and when i asked the last sprint about putting more data into the DTO someone told me not to do that and to leave it like this.

And the thing about pasword i dont know if I have understandit well. You can change data without having to change password. And if by this you mean that you should be asked the password to be able to chanege data, i dont think it should be necessary in this case.

Egoes98 committed:
Merge branch 'master' into User-Edit
